keyboards/printedpad/printedpad.c

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Noah Beidelman (@noahbei)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "printedpad.h"
+
+void board_init(void) {
+  SYSCFG->CFGR1 |= SYSCFG_CFGR1_I2C1_DMA_RMP;
+}
+
+uint8_t current_layer = 0;
+uint8_t current_display_mode = 0;
+
+bool key_pressed = false;
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+#ifdef OLED_ENABLE
+    if (record->event.pressed)
+        key_pressed = true;
+    else
+        key_pressed = false;
+#endif
+    switch(keycode) {
+        case LT(0, KC_NO):
+            // on tap
+            if (record->tap.count && record->event.pressed) {
+                current_layer = (current_layer + 1) % 4;
+                layer_move(current_layer);
+            }
+#ifdef OLED_ENABLE
+            // on hold
+            else if (record->event.pressed) {
+                current_display_mode = (current_display_mode + 1) % 3;
+            }
+#endif
```

Comment:
Instead of this workaround, you should use proper keycodes (eg. `MO(1)`, `MO(2)`, `MO(3)` etc.) and perhaps use the `layer_state_set_kb()` function to display the appropriate image based on layer.

Reply:
I used layer_move instead of keycodes like MO() because the intent is not to momentarily activate a layer, but completely move to it. Also having the variable current_layer helps with making the oled code easier to handle. I also used current_display_mode so that I can keep track of it to use inside of oled_task_user so that all of the oled code is triggered from there instead of layer_state_set_kb() (also easier to have the varible to work with for the oled code). I got the keycode intercept code straight from the docs, is it something that I really need to change because it works. - it is so that I can toggle display mode with hold and switch layers with tap.

Reply:
TO(x) on each layer would accomplish the same thing.  TO explicitly calls `layer_move`, in fact. 

Though, it looks like you're using lt to hack in tap-hold functionality here.

This block might be better as: 
```suggestion
        case LT(0, KC_NO):
            // on tap
            if (record->event.pressed) {
                if (record->tap.count)
                    if (get_highest_layer >= MAX_LAYER) {
                        layer_clear(); 
                    } else { 
                        layer_move(get_highest_layer(layer_state) + 1);
                    }
                }
#ifdef OLED_ENABLE
            // on hold
                else {
                    current_display_mode = (current_display_mode + 1) % 3;
                }
#endif
            }
```


But honestly, the whole thing ... feels excessively complex, and that a separate oled mode keycode would probably be a better option for this, in general.

Reply:
ya, it is a bit complex. I changed the code to what you had using layer_state (plus a missing parenthesis) and it works.